romanisim/tests/test_image.py

```diff
@@ -726,14 +722,22 @@ def test_inject_source_into_image():
         newramp * u.DN, ma_table,
         gain=1 * u.electron / u.DN, flat=1, dark=0)
 
-    # Test that all pixels outside of the psf of the injected source are equal to the original image
-    assert np.all(origimage.data[int(xpos) + psfXmax:int(xpos) - psfXmin, int(ypos) + psfYmax:int(ypos) - psfYmin].value
-                  == newimage[int(xpos) + psfXmax:int(xpos) - psfXmin, int(ypos) + psfYmax:int(ypos) - psfYmin].value)
+    # Create mask of PSF
+    nonzero = (sourcecounts.array != 0)
 
     # Test that all pixels inside of the psf of the injected source are different from the original image
-    assert np.any(origimage.data[int(xpos) - psfXmin:int(xpos) + psfXmax, int(ypos) - psfYmin:int(ypos) + psfYmax].value
-                  != newimage[int(xpos) - psfXmin:int(xpos) + psfXmax, int(ypos) - psfYmin:int(ypos) + psfYmax].value)
+    assert np.any((origimage.data.value != newimage.value), where=nonzero)
 
+    # Test that all pixels outside of the psf of the injected source are close to the original image
+    # Create array copies for all close comparison of non PSF-pixels
+    cleaned_origimage = copy.deepcopy(origimage.data.value)
+    cleaned_origimage[nonzero] = 1.0
+    cleaned_newimage = copy.deepcopy(newimage.value)
+    cleaned_newimage[nonzero] = 1.0
+
+    assert np.allclose(cleaned_origimage, cleaned_newimage, rtol=1e-05, atol=1e-08)
```

Comment:
Just trying to understand... is this equivalent to np.allclose(origimage.data.value[nonzero], newimage.data.value[nonzero])?

Reply:
I have cleaned this up to something similar.